extension-types.md

```diff
@@ -0,0 +1,102 @@
+
+# Arrow Extension Type definitions
+
+The memory layout specification provides an encoding for geometry types
+using the Arrow Columnar format; however, it does not provide a route by
+which columns can be recognized as geometry columns when interacting with
+Arrow bindings (e.g., Arrow C++, PyArrow, or the arrow package for R).
+Similarly, it does not provide a route by which geometry metadata (e.g.,
+coordinate reference system) can be attached at the column level.
+Whereas the GeoParquet metadata standard defines table-level metadata
+intended for geospatial-aware consumers of a file, this specification
+defines Array-level metadata intended for geospatial-aware consumers
+of individual Arrays (e.g., a library implementing a geospatial algorithm)
+or for non-geospaital aware consumers of a table containing such an
+Array (e.g., a compute engine that can select columns and/or filter rows).
+
+## Extension names
+
+When GeoArrow-encoded Arrays have the `ARROW:extension:name` metadata
+field set, it should be set to one of `geoarrow.point`, `geoarrow.linestring`,
+`geoarrow.polygon`, `geoarrow.multipoint`, `geoarrow.multilinestring`,
+`geoarrow.multipolygon`, or `geoarrow.wkb`. These names correspond
+to the memory layouts and value constraints described in
+[GeoArrow memory layout specification](format.md).
+
+The `ARROW:extension:name` and `ARROW:extension:metadata` metadata fields
+must also be set for child Arrays (i.e., extension types are nested).
+For the extension types noted above, the nesting is as follows:
+
+- `geoarrow.point`: `FixedSizeList<xy|xyz|xym|xyzm: double>[n_dim]`
+- `geoarrow.linestring`: `List<vertices: geoarrow.point>`
+- `geoarrow.polygon`: `List<rings: List<vertices: geoarrow.point>>`
+- `geoarrow.multipoint`: `List<points: geoarrow.point>`
+- `geoarrow.multilinestring`: `List<linestrings: geoarrow.linestring>`
+- `geoarrow.multipolygon`: `List<polygons: geoarrow.polygon>`
+- `geoarrow.wkb`: `Binary`, `LargeBinary`, or `FixedSizeBinary[fixed_size]`
+
+## Extension metadata
+
+When GeoArrow-encoded Arrays have the `ARROW:extension:metadata` metadata
+field set, it must be encoded using the same format as specified for
+[metadata in the C data interface](https://arrow.apache.org/docs/format/CDataInterface.html#c.ArrowSchema.metadata)
+with the constraint that the 32-bit integers that encode the length of
+each key and each value must be little-endian. This encoding provides
+the ability to store arbitrary key/value pairs where the key is a UTF-8
+encoded string and the value can be an arbitrary sequence of bytes.
+For all currently supported keys, the value will be interpreted as a
+UTF-8-encoded string.
+
+The following metadata keys are supported:
+
+- `crs_encoding`: Contains a UTF-8-encoded string describing how
+  the `crs` metadata field should be interpreted. Supported values of
+  `crs_encoding` are `"wkt2"` and `"projjson"`. If this key is omitted
```

Comment:
Should there be a discussion about whether to drop wkt2 support in GeoArrow as well? For one, a non-CRS-aware implementation would not be able to write this to GeoParquet